src/devices/Mcp9808/Mcp9808.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+using Iot.Units;
+
+namespace Iot.Device.Mcp9808
+{
+    /// <summary>
+    /// Microchip's MCP9808 I2C Temp sensor
+    /// </summary>
+    public class Mcp9808 : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// MCP9808 I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x18;
+
+        #region prop
+
+        /// <summary>
+        /// MCP9808 Temperature
+        /// </summary>
+        public Temperature Temperature { get => Temperature.FromCelsius(GetTemperature()); }
+
+        private bool _disable;
+
+        /// <summary>
+        /// Disable MCP9808
+        /// </summary>
+        public bool Disabled
+        {
+            get
+            {
+                return _disable;
+            }
+            set
+            {
+                SetShutdown(value);
+                _disable = value;
+            }
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Creates a new instance of the MCP9808
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device used for communication.</param>
+        public Mcp9808(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice;
+
+            Disabled = false;
+        }
+
+        /// <summary>
+        /// Checks if the device is a MCP9808
+        /// </summary>
+        /// <returns>True if device has been correctly detected</returns>
+        public bool Init()
```

Comment:
I think this should be private, I'd call it in the constructor and throw if doesn't match